File: c++/Route.cpp
#include <iostream>
#include "Route.h"

using namespace std;

Route::Route(int routeId, string source, string destination,double distance, string estimatedTime)
{
    this->routeId = routeId;
    this->source = source;
    this->destination = destination;
    this->distance = distance;
    this->estimatedTime = estimatedTime;
}

// Displays route details including ID, source, destination,
// distance, and estimated time
void Route::displayRouteDetails()
{
    cout << "\n----------- Route Details -----------" << endl;
    cout << "Route ID : " << routeId << endl;
    cout << "Source : " << source << endl;
    cout << "Destination : " << destination << endl;
    cout << "Distance : " << distance << " km" << endl;
    cout << "Estimated Time : " << estimatedTime << endl;
}

File: python/Route.py
class Route:

    def __init__(self, routeId, source, destination, distance, estimatedTime):
        self.__routeId = routeId
        self.__source = source
        self.__destination = destination
        self.__distance = distance
        self.__estimatedTime = estimatedTime

    # Displays route details including ID, source, destination,
    # distance, and estimated time
    def displayRouteDetails(self):

        print("\n----------- Route Details -----------")
        print("Route ID :", self.__routeId)
        print("Source :", self.__source)
        print("Destination :", self.__destination)
        print("Distance :", self.__distance, "km")
        print("Estimated Time :", self.__estimatedTime)